src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -0,0 +1,378 @@
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import time
+import json
+import os
+import sas
+import sas_objects
+import constraint_testcase
+from util import winnforum_testcase, configurable_testcase, writeConfig, \
+  loadConfig, generateCpiRsaKeys, generateCpiEcKeys, convertRequestToRequestWithCpiSignature
+from reference_models.aggregate_interference import agg_interf
+from reference_models.pre_iap import pre_iap
+from reference_models.iap import iap
+
+
+class MultiConstraintProtectionTestcase(constraint_testcase.ConstraintTestcase):
+
+    def setUp(self):
+        self._sas, self._sas_admin = sas.GetTestingSas()
+        self._sas_admin.Reset()
+  
+    def tearDown(self):
+        pass
+  
+    def generate_MCP_1_default_config(self, filename):
+        """ Generates the WinnForum configuration for MCP.1. """
+        # Load CBSD
+        device_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'device_a.json')))
+        device_2 = json.load(
+          open(os.path.join('testcases', 'testdata', 'device_b.json')))
+        device_3 = json.load(
+          open(os.path.join('testcases', 'testdata', 'device_c.json')))
+        device_4 = json.load(
+          open(os.path.join('testcases', 'testdata', 'device_d.json')))
+        device_5 = json.load(
+          open(os.path.join('testcases', 'testdata', 'device_e.json')))
+  
+        # Load Grant requests
+        grant_request_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+        grant_request_2 = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_1.json')))
+        grant_request_3 = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_2.json')))
+        grant_request_4 = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_3.json')))
+      
+        # Load GWPZ Record
+        gwpz_record_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'gwpz_record_0.json')))
+  
+        # Load FSS record
+        fss_record_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'fss_record_0.json')))
+  
+        # Load ESC record
+        esc_record_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'esc_sensor_record_0.json')))
+     
+        # Load PPA record
+        ppa_record_1 = json.load(
+          open(os.path.join('testcases', 'testdata', 'ppa_record_0.json'))) 
+        # Registration configuration for iteration 1
+        cbsd_records_iter_1 = {'registrationRequests': [device_1, device_2],
+                              'grantRequests': [grant_request_1, grant_request_2]}
+        cbsd_records_iter_2 = {'registrationRequests': [device_3, device_4],
+                              'grantRequests': [grant_request_3, grant_request_4]}
+        protected_entities_iter_1 = {'PPA': [ppa_record_1], 'ESC': [esc_record_1]}
```

Comment:
In order to inject a PPA, we also have to inject the corresponding PAL records.

Reply:
Updated by adding PalAndPpa field in protectedEntities which has a set of   pal and ppa records